Skops announcement blog

merveenoyan opened the pull request:
This PR contains skops announcement blog.

Might be edited after first release, I'll also add an example repository of how the pushed model looks like, but let me know opinions on the initial version of this blog post.

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+
+```python
+import pickle
+from tempfile import mkstemp, mkdtemp
````
Is there a particular reason why both `mkstemp` and `mkdtemp` are used? From reading the docs, it would seem that `mkdtemp` is enough.

merveenoyan replied:
I think I'll avoid using both for this tutorial later on.

merveenoyan replied:
I put names specifically to make it intuitive 🙂

BenjaminBossan commented on skops-library.md:
```diff
@@ -0,0 +1,147 @@
+import pickle
+from tempfile import mkstemp, mkdtemp
+from skops import hub_utils, card
+import matplotlib.pyplot as plt
```
```suggestion
import sklearn
import matplotlib.pyplot as plt
```

`sklearn.__version__` is used below

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+)
+```
+
+`init` not only saves the model, but also creates a configuration file containing the specifications of the environment model is trained. 
````
I think the "saves the model" part could be confusing to readers, since calling it neither saves the model on disk, nor on Hub.

merveenoyan replied:
Changed it 🙂

BenjaminBossan commented on skops-library.md:
```diff
@@ -0,0 +1,147 @@
+
+`init` not only saves the model, but also creates a configuration file containing the specifications of the environment model is trained. 
+
+We will now create a model card. The content of the model card is determined by a jinja template. By default it uses [this template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md). The default template consists of a markdown part and a metadata section. The keys to the metadata section is defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata ...
```
Perhaps add a link for those readers who don't know what model cards are?

BenjaminBossan replied:
```suggestion
We will now create a model card. The content of the model card is determined by a jinja template. By default it uses [this template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md). The default template consists of a markdown part and a metadata section. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
```

BenjaminBossan commented on skops-library.md:
```diff
@@ -0,0 +1,147 @@
+
+We will now create a model card. The content of the model card is determined by a jinja template. By default it uses [this template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md). The default template consists of a markdown part and a metadata section. The keys to the metadata section is defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata ...
+
+We can add information and metadata using `add`. You can also pass a custom template's path to be used in model card using `add`.
```
```suggestion
We can add information and metadata using `add` method on the Skops `Card` object. You can also pass a custom template's path to be used in model card using `add`.
```

Also, maybe let's not mention the custom template for now? Reason being that we discussed changing the API for that and it's not shown below anyway.

BenjaminBossan commented on skops-library.md:
```diff
@@ -0,0 +1,147 @@
+disp.plot()
+
+# save the plot
+plt.savefig(f"{local_repo}/confusion_matrix.png")
```
The example uses a `Path` in other places, let's be consistent.
```suggestion
plt.savefig(Path(local_repo) / "confusion_matrix.png")
```

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+
+```python
+import pickle
+from tempfile import mkstemp, mkdtemp
````
```suggestion
from pathlib import Path
from tempfile import mkstemp, mkdtemp
```

`Path` is used below.

merveenoyan replied:
added it

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+
+Let's save the model card in the local repository. 
+```python
+model_card.save((Path(local_repo) / "README.md"))
````
```suggestion
model_card.save(Path(local_repo) / "README.md")
```

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+model_card.save((Path(local_repo) / "README.md"))
+```
+
+We can now push the repository to Hugging Face Hub. For this, we will use push from hub_utils.
````
```suggestion
We can now push the repository to Hugging Face Hub. For this, we will use `push` from `hub_utils`.
```

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+```python
+# if the repository doesn't exist remotely on the Hugging Face Hub, it will be created when we set create_remote to True
+hub_utils.push(
+    repo_id="skops-user/my-awesome-model",
````
Maybe we can define `repo_id` as a variable and reference it here and everywhere else it is used. Otherwise, a user who copies the code needs to adapt it in multiple places to replace the `repo_id` with their own.

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+Models can be serialized for reproducibility with skops. You can later push your model to Hugging Face Hub as well.
+
+```python
+import pickle
````
```suggestion
import os
import pickle
```

`os` is used below.

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+print(os.listdir(repo_copy))
+```
+
+The repository also contains the model configuration as well as requirements of the environment.
````
What does the "also" refer to?

merveenoyan replied:
Repo contains model, and also configuration as well as requirements of the env.

BenjaminBossan replied:
Then how about: "Apart from the model file, the repository contains the model configuration and the requirements of the environment."

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,147 @@
+print(json.dumps(hub_utils.get_config(path=repo_copy), indent=2))
+```
+
+If the requirements of your project has changed, you can use `update_env` to update the environment.
````
```suggestion
If the requirements of your project have changed, you can use `update_env` to update the environment.
```

BenjaminBossan reviewed (commented):
This is a good write up, thanks Merve.

I have found a few issues, please take a look at my comments. Apart from those, I would actually propose to wait working on the blog post until the code is actually ready. The reason being that we may still change a few things and then that would require updating the blog post as well. I believe it's more efficient if we have the code ready.

osanseviero commented on skops-library.md:
```diff
@@ -0,0 +1,149 @@
+title: "Introducing Skops"
+thumbnail: /blog/assets/88_skops_library/introducing_skops.png
+---
+
```
You will need to add this to https://github.com/huggingface/blog/blob/main/_blog.yml for it to show

osanseviero commented on skops-library.md:
```diff
@@ -0,0 +1,149 @@
+---
```
The file name is the url, I think just `skops` is good, that way the url is hf.co/blog/skops

osanseviero commented on skops-library.md:
```diff
@@ -0,0 +1,149 @@
+</h1>
+
+<div class="blog-metadata">
+    <small>Published N/A.</small>
```
To update before merge

osanseviero commented on skops-library.md:
```diff
@@ -0,0 +1,149 @@
+
+## Introducing Skops
+
+At Hugging Face, we are working on tackling various problems in open-source machine learning, including, hosting models securely and openly, enabling reproducibility, explainability and collaboration. We are thrilled to introduce you to our new library: Skops! With skops, you can host your sklearn models on Hugging Face Hub, create model cards with better reproducibility and collaborate with othe ...
```
Nit: Please capitalize Skops consistently (also in other sections)

```suggestion
At Hugging Face, we are working on tackling various problems in open-source machine learning, including, hosting models securely and openly, enabling reproducibility, explainability and collaboration. We are thrilled to introduce you to our new library: Skops! With Skops, you can host your sklearn models on the Hugging Face Hub, create model cards with better reproducibility and collaborate with others. 
```

osanseviero replied:
I think model cards for documentacion and reproducibility are two different features :)

osanseviero commented on skops-library.md:
```diff
@@ -0,0 +1,149 @@
+
+At Hugging Face, we are working on tackling various problems in open-source machine learning, including, hosting models securely and openly, enabling reproducibility, explainability and collaboration. We are thrilled to introduce you to our new library: Skops! With skops, you can host your sklearn models on Hugging Face Hub, create model cards with better reproducibility and collaborate with othe ...
+
+Models can be serialized for reproducibility with skops. You can later push your model to Hugging Face Hub as well.
```
```suggestion
Models can be serialized for reproducibility with skops. You can later push your model to the Hugging Face Hub as well.
```

osanseviero reviewed (commented):
I did not review the code as per comments above it will change, so just left some nits and suggestions :)

BenjaminBossan commented on skops-library.md:
```diff
@@ -0,0 +1,190 @@
+disp.plot()
+
+# save the plot
+plt.savefig(f"{local_repo}/confusion_matrix.png")
```
```suggestion
plt.savefig(Path(local_repo) / "confusion_matrix.png")
```
for consistency?

osanseviero commented:
@merveenoyan I see unresolved comments from last time (both from @BenjaminBossan and mine). It would be great if you fix them + marked as resolved (e.g. name of file, blog.yml, etc)

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,190 @@
+print(json.dumps(hub_utils.get_config(path=download_repo), indent=2))
+```
+
+If the requirements of your project have changed, you can use `update_env` to update the environment.
````
Should we remove this? The function currently doesn't do anything.

merveenoyan replied:
I thought to leave it like that in case if user wants to run it, but sure 😅

BenjaminBossan commented on skops-library.md:
````diff
@@ -0,0 +1,190 @@
+hub_utils.update_env(path=local_repo, requirements=["scikit-learn"])
+```
+
+We have prepared two notebooks to demonstrate how to save your models and use model card utilities, you can find them at resources section below.
````
> notebooks

Not really "notebooks", maybe just "examples"?

merveenoyan replied:
sure!

BenjaminBossan reviewed (commented):
Content-wise looks good to me, well written. I have a few minor comments, please take a look.

merveenoyan commented:
@osanseviero I left your comments intentionally because the PR is far from to be merged yet, wanted to update the PR right before the merger.

osanseviero commented:
I see multiple open comments from the first review from @BenjaminBossan two weeks ago. It's a bit harder to review the PR in the current state if there are sections that based on previous reviews will change (or if they are fixed/resolved already, you can mark the comments as resolved 😄). This helps reduce the number of review iterations.

Btw there are some comments such as renaming the file or adding to blog.yml that don't need to be done right before the merge

merveenoyan commented:
@osanseviero you're right, I missed a comment from Benjamin. I addressed your comments on my last commit.

osanseviero commented on _blog.yml:
```diff
@@ -1087,3 +1087,12 @@
+  author: merve
+  thumbnail: /blog/assets/88_skops_library/introducing_skops.png
+  date: August 15, 2022
+  tags:
```
open-source-collab might be a useful tag here, that way it will show when you go to https://huggingface.co/blog?tag=open-source-collab

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+---
+title: "Introducing Skops"
+thumbnail: /blog/assets/88_skops_library/introducing_skops.png
```
I feel the "Introducing" being orange with a orangish background might be a bit hard to read. No strong opinions though as it's not my area of expertise. Also cc @Reeray on this

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+</div>
+
+<div class="author-card">
+    <a href="/merve">
```
Should we add Adrin and Benjamin as coauthors on this?

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+At Hugging Face, we are working on tackling various problems in open-source machine learning, including, hosting models securely and openly, enabling reproducibility, explainability and collaboration. We are thrilled to introduce you to our new library: Skops! With Skops, you can host your sklearn models on the Hugging Face Hub, create model cards for model documentation and collaborate with othe ...
+
+Let's go through an end-to-end example: train a model first, and see step-by-step how to leverage Skops for sklearn in production.
+
```
For code heavy blog posts, we usually also add companion notebooks that people can open to try out the same code. The notebooks are at https://github.com/huggingface/blog/tree/main/notebooks See https://huggingface.co/blog/fastai as an example

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix
+from sklearn.datasets import load_breast_cancer
+from sklearn.ensemble import HistGradientBoostingClassifier
+from sklearn.experimental import enable_halving_search_cv  # noqa
```
Would it make more sense to use a much simpler model? Maybe one of the main examples of sklearn such as iris from https://scikit-learn.org/stable/tutorial/basic/tutorial.html? My concern with looking at this snippet is:

* It's using cross validation, param grid and other things which make the code more complex. If you're introducing a new library to people and on top of that you show different sklearn features they might not be aware of, they have a higher cognitive overload. Many people might not know `HistGradientBoostingClassifier`, but  `svm.SVC` is probably better known out there.
* Seeing an experimental `sklearn` feature (`enable_halving_search_cv`) here is unexpected to me (and unnecessary afaik)
* Having a shorter, more concise and minimal snippet will make it less likely for people to walk away from reading the whole thing

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+import pickle
+from skops import hub_utils, card
+import sklearn
+import matplotlib.pyplot as plt
```
`plt` is not used in this snippet. For blog posts, usually it's better to do the imports in the code snippets where they are used.  Same for `pickle` and `Path`and few others

osanseviero commented on skops.md:
````diff
@@ -0,0 +1,180 @@
+).fit(X_train, y_train)
+```
+
+We will first save our model and initialize the repository. The model name and the format is insignificant, we will save our model as `example.pkl` for simplicity. `init` creates a repository containing the model in the given path and the configuration file containing the specifications of the environment the model is trained in. The data and the task passed to the `init` will help Hugging Face H ...
````
> The model name and the format is insignificant, we will save our model as `example.pkl` for simplicity.

This sounds a bit strange. Should it be something like

> You can use any model filename you want. You can also use your serializing method of preference, like pickle or joblib. (?)

osanseviero replied:
```suggestion
We will first save our model and initialize the repository. The model name and the format is insignificant, we will save our model as `example.pkl` for simplicity. `hub_utils .init` creates a Git repository containing the model in the given path and the configuration file containing the specifications of the environment the model is trained in. The data and the task passed to the `init` will help Hugging Face Hub enable the inference widget on the model page as well as discoverability features to find the model.
```

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+    pickle.dump(model, file=f)
+
+# we will now initialize a local repository
+hub_utils.init(
+    model=model_path, 
+    requirements=[f"scikit-learn={sklearn.__version__}"], 
+    dst=local_repo,
+    task="tabular-classification",
+    data=X_test,
+)
```
Some questions I had from looking at above
* Should `model` receive a path or a model. If it's a path, should the parameter name change to `model_path`?
* It's not very clear from the previous paragraph if the repo is locally or also pushed automatically. Should we clarify it's a local clone of a remote repo?

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+
+The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task solved here.
+
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
```
```suggestion
We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section, which is a `yaml` section at the top. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
```

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task solved here.
+
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
+The content of the model card is determined by a jinja template that has:
```
People might not know what Jinja is and afaik for most use cases people don't need to know this, so maybe we can just remove "jinja" from the sentence. WDYT?

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
+The content of the model card is determined by a jinja template that has:
+- yaml section on top for metadata (e.g. model license, library name and more),
+- markdown section with free text and slots to be filled (e.g. simple description of the model),
```
```suggestion
- markdown section with free text and sections to be filled (e.g. simple description of the model),
```

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+Following sections are extracted by Skops to fill in the model card:
+- Hyperparameters of the model,
+- Interactive plot of the model,
+- For metadata, library name and task identifier (e.g. tabular-classification) are filled.
+We will walk you through how to programmatically pass information to fill the model card. Please take a look at the [default template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md) used by Skops to see what the template expects.
```
```suggestion
- For metadata, library name and task identifier (e.g. tabular-classification) are filled.

We will walk you through how to programmatically pass information to fill the model card. Please take a look at the [default template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md) used by Skops to see what the template expects.
```

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+# create the card
+model_card = card.Card(model)
+
+# create some information we want to pass to the card
```
```suggestion
# define some metadata to be added to the card 
```

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+# create some information we want to pass to the card
+license = "mit"
+limitations = "This model is not ready to be used in production."
+model_description = (
```
Why is this with parenthesis?

osanseviero commented on skops.md:
```diff
@@ -0,0 +1,180 @@
+
+# the plot will be written to the model card under the name confusion_matrix
+# we pass the path of the plot itself
+model_card.add_plot(confusion_matrix="confusion_matrix.png")
```
This is very cool! 🔥

osanseviero commented on skops.md:
````diff
@@ -0,0 +1,180 @@
+model_card.save(Path(local_repo) / "README.md")
+```
+
+We can now push the repository to Hugging Face Hub. For this, we will use `push` from `hub_utils`. Hugging Face Hub requires tokens for authentication, therefore you need to pass your token.
````
Alternatively they can log in with `huggingface_hub login` or `notebook_login`

osanseviero commented on skops.md:
````diff
@@ -0,0 +1,180 @@
+We can now push the repository to Hugging Face Hub. For this, we will use `push` from `hub_utils`. Hugging Face Hub requires tokens for authentication, therefore you need to pass your token.
+
+```python
+token = os.environ["HF_HUB_TOKEN"]
````
Isn't the token automatically obtained within skops if it's set in an environment variable?

osanseviero commented on skops.md:
````diff
@@ -0,0 +1,180 @@
+```python
+hub_utils.update_env(path=local_repo, requirements=["scikit-learn"])
+```
+
````
One very cool thing is having the widget, so maybe it's worth highlighting it at the end and having a screenshot

osanseviero reviewed (commented):
Very cool blog post 🔥 amazing work!

merveenoyan commented:
- I added @adrinjalali and @BenjaminBossan as authors (let me know if you don't want me to use your hf.co avatars)
- I added metadata from config (completely forgot previously)
- I put imports in their own corresponding snippets as @osanseviero suggested
- Put link to the repository pushed with the code that's used in this example
- Used `DecisionTreeClassifier` instead for simplicity and changed model description accordingly
- Added evaluation (though it's not merged yet, it's approved and is likely to be merged soon) details and results
- Flipped background of the image for banner
- Added screenshot of enabled widget and explanation
- Changed tags of blog in `_blog.yml`
- Put details on how to pass token (notebook login and CLI)
LMK if there's anything else you'd wish to add

adrinjalali commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+- markdown section with free text and sections to be filled (e.g. simple description of the model),
+Following sections are extracted by `skops` to fill in the model card:
+- Hyperparameters of the model,
+- Interactive plot of the model,
```
```suggestion
- Interactive diagram of the model,
```

adrinjalali commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+Following sections are extracted by `skops` to fill in the model card:
+- Hyperparameters of the model,
+- Interactive plot of the model,
+- For metadata, library name and task identifier (e.g. tabular-classification) are filled.
```
```suggestion
- For metadata, library name and task identifier (e.g. tabular-classification), and information required by the inference widget are filled.
```

adrinjalali commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+- Interactive plot of the model,
+- For metadata, library name and task identifier (e.g. tabular-classification) are filled.
+
+We will walk you through how to programmatically pass information to fill the model card. Please take a look at the [default template](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md) used by `skops` to see what the template expects.
```
We shouldn't send people to the repo, here's the documentation of the default template: https://skops.readthedocs.io/en/latest/model_card.html, if something's missing there, should be added.

adrinjalali reviewed (commented):
Thanks for working on this @merveenoyan

BenjaminBossan commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+limitations = "This model is not ready to be used in production."
+model_description = "This is a DecisionTreeClassifier model trained on breast cancer dataset."
+model_card_authors = "skops_user"
+get_started_code = "import pickle \nwith open(dtc_pkl_filename, 'rb') as file: \nclf = pickle.load(file)"
```
```suggestion
get_started_code = "import pickle \nwith open(dtc_pkl_filename, 'rb') as file: \n    clf = pickle.load(file)"
```
Same comment as I did previously, this is needed or else the block is not indented.

merveenoyan commented:
@stevhliu Can you take a look at our blog? 👀

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+model = DecisionTreeClassifier().fit(X_train, y_train)
+```
+
+You can use any model filename you want. You can also use any serializing method that you prefer, like `pickle` or `joblib`. At the moment our backend uses `joblib` to load the model. `hub_utils .init` creates a local folder containing the model in the given path and the configuration file containing the specifications of the environment the model is trained in. The data and the task passed to th ...
````
```suggestion
You can use any model filename and serialization method, like `pickle` or `joblib`. At the moment, our backend uses `joblib` to load the model. `hub_utils.init` creates a local folder containing the model in the given path, and the configuration file containing the specifications of the environment the model is trained in. The data and the task passed to the `init` will help Hugging Face Hub enable the inference widget on the model page as well as discoverability features to find the model.
```

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+)
+```
+
+The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task solved here.
````
I think breaking this up into a list makes it more readable.

stevhliu replied:
```suggestion
The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task to be solved here.
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+
+The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task solved here.
+
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section, which is a `yaml` section at the top. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
```
```suggestion
We will now create the model card. The card should match the expected Hugging Face Hub format: a markdown part and a metadata section, which is a `yaml` section at the top. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for the discoverability of the models. 
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+The repository now contains the serialized model and the configuration file. The configuration contains the features of the model, the requirements of the model, an example input taken from `X_test` that we've passed, name of the model file and name of the task solved here.
+
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section, which is a `yaml` section at the top. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
+The content of the model card is determined by a template that has:
```
```suggestion
The content of the model card is determined by a template that has a:
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+
+We will now create the model card. The card should fit the format that Hugging Face Hub expects it to be: it consists of a markdown part and a metadata section, which is a `yaml` section at the top. The keys to the metadata section are defined [here](https://huggingface.co/docs/hub/models-cards#model-card-metadata) and are used for discoverability of the models. 
+The content of the model card is determined by a template that has:
+- yaml section on top for metadata (e.g. model license, library name and more),
```
You can also remove the commas at the end of the list :)

```suggestion
- `yaml` section on top for metadata (e.g. model license, library name, and more)
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+The content of the model card is determined by a template that has:
+- yaml section on top for metadata (e.g. model license, library name and more),
+- markdown section with free text and sections to be filled (e.g. simple description of the model),
+Following sections are extracted by `skops` to fill in the model card:
```
```suggestion
The following sections are extracted by `skops` to fill in the model card:
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+Following sections are extracted by `skops` to fill in the model card:
+- Hyperparameters of the model,
+- Interactive diagram of the model,
+- For metadata, library name and task identifier (e.g. tabular-classification), and information required by the inference widget are filled.
```
```suggestion
- For metadata, library name, task identifier (e.g. tabular-classification), and information required by the inference widget are filled.
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+- Interactive diagram of the model,
+- For metadata, library name and task identifier (e.g. tabular-classification), and information required by the inference widget are filled.
+
+We will walk you through how to programmatically pass information to fill the model card. You can check out our documentation on default template provided by `skops` and it's sections [here](https://skops.readthedocs.io/en/latest/model_card.html) to see what the template expects and see how the template itself looks like [here](https://github.com/skops-dev/skops/blob/main/skops/card/default_templ ...
```
```suggestion
We will walk you through how to programmatically pass information to fill the model card. You can check out our documentation on the default template provided by `skops`, and its sections [here](https://skops.readthedocs.io/en/latest/model_card.html) to see what the template expects and what it looks like [here](https://github.com/skops-dev/skops/blob/main/skops/card/default_template.md).
```

stevhliu commented on skops.md:
```diff
@@ -0,0 +1,200 @@
+
+We will walk you through how to programmatically pass information to fill the model card. You can check out our documentation on default template provided by `skops` and it's sections [here](https://skops.readthedocs.io/en/latest/model_card.html) to see what the template expects and see how the template itself looks like [here](https://github.com/skops-dev/skops/blob/main/skops/card/default_templ ...
+
+You can create the model card by instantiating `Card` class from `skops`. During serialization of the model, task name and the library name are written to configuration file. This information is needed in card's metadata as well, so you can use `metadata_from_config` method to extract the metadata needed from configuration and pass it to the card while creation. You can add information and metada ...
```
```suggestion
You can create the model card by instantiating the `Card` class from `skops`. During model serialization, the task name and library name are written to the configuration file. This information is also needed in the card's metadata, so you can use the `metadata_from_config` method to extract the metadata from the configuration file and pass it to the card when you create it. You can add information and metadata using `add`.
```

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+model_card.metadata.license = "mit"
+```
+
+We will now evaluate the model and pass details on the evaluation process. To pass the details about the evaluation method, we can use `add`. We can pass the metrics using `add_metrics`. The metrics that we'll pass will be parsed to a table. 
````
```suggestion
We will now evaluate the model and add a description of the evaluation method with `add`. The metrics are added by `add_metrics`, which will be parsed into a table. 
```

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+model_card.save(Path(local_repo) / "README.md")
+```
+
+We can now push the repository to Hugging Face Hub. For this, we will use `push` from `hub_utils`. Hugging Face Hub requires tokens for authentication, therefore you need to pass your token. You can use `notebook_login` if you're logging in from notebook, or you can use `huggingface-cli login` if you're logging in from CLI, you can also pass your token directly.
````
```suggestion
We can now push the repository to the Hugging Face Hub. For this, we will use `push` from `hub_utils`. Hugging Face Hub requires tokens for authentication, therefore you need to pass your token in either  `notebook_login` if you're logging in from a notebook, or `huggingface-cli login` if you're logging in from the CLI.
```

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+)
+```
+
+Once we push the model to the Hub, anyone can use it, unless the repository is private. You can download the models using `download`. Apart from the model file, the repository contains the model configuration and the requirements of the environment."
````
```suggestion
Once we push the model to the Hub, anyone can use it unless the repository is private. You can download the models using `download`. Apart from the model file, the repository contains the model configuration and the environment requirements.
```

stevhliu commented on skops.md:
````diff
@@ -0,0 +1,200 @@
+```
+
+You can see the example repository pushed with above code [here](https://huggingface.co/scikit-learn/skops-blog-example).
+We have prepared two examples to show how to save your models and use model card utilities, you can find them at resources section below.
````
```suggestion
We have prepared two examples to show how to save your models and use model card utilities. You can find them in the resources section below.
```

stevhliu reviewed (commented):
This is really cool, love how easy you made it to create a model card from the configuration file! 🤗

merveenoyan commented:
@stevhliu I committed rewording suggestions and put config into list as you asked. Let me know if there's any other change you'd like me to do.

stevhliu commented:
Awesome, looks good to me!

merveenoyan commented:
@stevhliu can you merge if you like it? 😅

merveenoyan commented:
@adrinjalali  the hosted inference image doesn't seem to work, I'm currently off, I checked it out but I can't seem to find the problem. can you do me a favor and check it out?

adrinjalali commented:
This should fix it: https://github.com/huggingface/blog/pull/468
